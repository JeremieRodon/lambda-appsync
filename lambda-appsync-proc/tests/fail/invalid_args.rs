use lambda_appsync::{appsync_lambda_main, appsync_operation, AppsyncError, ID};

appsync_lambda_main!("../../../../schema.graphql");

// Wrong argument type (i32 instead of String)
#[appsync_operation(mutation(createPlayer))]
async fn create_player(name: i32) -> Result<Player, AppsyncError> {
    Ok(Player {
        id: ID::new(),
        name: name.to_string(),
        team: Team::Rust,
    })
}
