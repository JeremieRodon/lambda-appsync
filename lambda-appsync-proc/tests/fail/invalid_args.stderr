error[E0631]: type mismatch in function arguments
 --> tests/fail/invalid_args.rs:7:10
  |
6 | #[appsync_operation(mutation(createPlayer))]
  | --------------------------------------------
  | |
  | found signature defined here
  | required by a bound introduced by this call
7 | async fn create_player(name: i32) -> Result<Player, AppsyncError> {
  |          ^^^^^^^^^^^^^ expected due to this
  |
  = note: expected function signature `fn(std::string::String) -> _`
             found function signature `fn(i32) -> _`
note: required by a bound in `create_player::without_event::check_signature`
 --> tests/fail/invalid_args.rs:3:1
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `check_signature`
  = note: this error originates in the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider wrapping the function in a closure
  |
7 | async fn |arg0: std::string::String| create_player(/* i32 */)(name: i32) -> Result<Player, AppsyncError> {
  |          +++++++++++++++++++++++++++              +++++++++++
